src/storage/src/hummock/event_handler/uploader/mod.rs

```diff
@@ -807,7 +826,20 @@ impl UnsyncData {
             .get_mut(&instance_id)
             .expect("should exist");
         let epoch = instance_data.local_seal_epoch(next_epoch);
-        self.epochs.insert(next_epoch, ());
+        // When drop/cancel a streaming job, for the barrier to stop actor, the
+        // local instance will call `local_seal_epoch`, but the `next_epoch` won't be
+        // called `start_epoch` because we have stopped writing on it.
+        if !table_data.unsync_epochs.contains_key(&next_epoch) {
```

Comment:
is it possible that `table_data` for the dropping/cancelling streaming job tables are absent causing L823 to panic?

Reply:
It's unlikely to happen. 

Let's say the barrier that stops the actor has epoch{curr=3, prev=2}. Before calling `sync` on `epoch.prev`, we must have called `start_epoch(epoch.prev)`. Before calling this `local_seal_epoch(epoch.curr)`, this barrier won't be collected, and we cannot call `sync(epoch.prev)`. Before `sync(epoch.prev)`, the `unsync_epochs` must contain `epoch.prev`, and therefore the `table_data` is non-empty and won't be dropped, and then in L823 the table won't be absent.

Reply:
Got it. Thanks for the explanation.